Assets/Scripts/Enemy/PatrollingState.cs

```diff
@@ -0,0 +1,27 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class PatrollingState : States {
+
+	protected override void Awake()
+	{
+		base.Awake();
+
+	}
+	public override void OnEnterState()
+	{
+		base.OnEnterState();
+		enemyView.ResumePatrol();
+	}
+
+	public override void OnExitState()
+	{
+		base.OnExitState();
+	}
+
+	private void OnCollisionEnter(Collision collision)
+	{
+		enemyView.ChangeState(enemyView.chasingState);
```

Comment:
you should check who you are colliding with and then decide what you want to do - on every collision, this will change state to chasing state